src/nox_poetry/poetry.py

```diff
@@ -69,22 +93,38 @@ def config(self) -> Config:
             self._config = Config(Path.cwd())
         return self._config
 
-    def export(self) -> str:
+    def export(
+        self,
+        groups: Optional[List[str]] = None,
+    ) -> str:
         """Export the lock file to requirements format.
 
+        Args:
+            groups: optional list of poetry depedency groups to --only install.
+
         Returns:
             The generated requirements as text.
 
         Raises:
             CommandSkippedError: The command `poetry export` was not executed.
         """
-        output = self.session.run_always(
+        args = [
             "poetry",
             "export",
             "--format=requirements.txt",
-            "--dev",
             *[f"--extras={extra}" for extra in self.config.extras],
             "--without-hashes",
+        ]
+
+        if groups:
+            args.extend(f"--only={group}" for group in groups)
+        elif self.config.is_compatible_with_group_deps():
+            args.append("--with=dev")
+        else:
+            args.append("--dev")
```

Comment:
Why is the check on `self.config.is_compatible_with_group_deps()` only for `--with`?  Isn't `--only` also only available for `poetry>=1.2`?

Reply:
Reading this again, it looks like I misunderstood your comment. Ok. You're asking why the first `if groups:` clause doesn't also check for `self.config.is_compatible_with_group_deps()` since that `if groups:` clause uses the `--only` flag which is only available for poetry v>=1.2.

The reason is because earlier in the code we have a [check](https://github.com/cjolowicz/nox-poetry/pull/1080/files#diff-c0353ddf703d4a61f7318415e57e355f3a14987fd0a4ed4c0b5ff94db46acc01R257) within `_PoetrySession.export_requirements` that makes sure that if you're using "groups" that you're using the right version of poetry. And there's a [unit test](https://github.com/cjolowicz/nox-poetry/pull/1080/files#diff-04725095aad4f11a99425d6be5f1fefd2b851078c1bc8b963a5a1d569131655eR245) to ensure that.

Reply:
Got it.  Thanks!